### ISSUE TITLE
No longer needed?

### ISSUE THREAD

**Issue by karlhorky** [JLongley/gatsby-remark-external-links#22]
**Edit:** As @afucher pointed out below, it may still be needed for `gatsby-transformer-remark`

---

[As of `@mdx-js/mdx` v1.6.1](https://github.com/mdx-js/mdx/pull/1048), `gatsby-remark-external-links` is no longer needed - you can use [`remark-external-links`](https://www.npmjs.com/package/remark-external-links) by itself 🎉

More details here: https://github.com/gatsbyjs/gatsby/issues/22719#issuecomment-624605087

I would recommend removing this from the [Gatsby Plugins](https://www.gatsbyjs.org/docs/plugins/) page and/or mark it as "Deprecated", and point at [`remark-external-links`](https://www.npmjs.com/package/remark-external-links).

**Comment by afucher** [comment]
@karlhorky but for `gatsby-transformer-remark` we still need `gatsby-remark-external-links`, right?

**Comment by karlhorky** [comment]
Good question, have you tried it out in an app?

I tried quickly in [this CodeSandbox](https://codesandbox.io/s/hopeful-field-9p9j4?file=/gatsby-config.js) and it appears to be broken:

![Screen Shot 2020-05-15 at 11 11 55](https://user-images.githubusercontent.com/1935696/82033546-2ac7f200-969d-11ea-9711-e21d0c3d6c1a.png)

It looks like `gatsby-transformer-remark` uses [`remark-parse`](https://github.com/remarkjs/remark/tree/master/packages/remark-parse) internally, which I assume should support standard Remark plugins.

@pieh @wooorm maybe this incompatibility between `gatsby-transformer-remark` and Remark plugins is worth another PR, this time to `gatsby-transformer-remark`?

**Comment by karlhorky** [comment]
Ah it looks like publishing additional "wrapper" npm packages to wrap every remark plugin is the recommended way of doing things 😞 At least in 2018 it was like this.

**References:**

- @KyleAMathews in https://github.com/gatsbyjs/gatsby/issues/2322#issuecomment-381345998
- @pieh in https://github.com/gatsbyjs/gatsby/issues/9103#issuecomment-429670139
- More background https://github.com/gatsbyjs/gatsby/issues/8009

So @wooorm I guess the process of writing extra wrapper packages you described the downsides about in https://github.com/gatsbyjs/gatsby/issues/22719#issuecomment-622305670 is actually recommended 😞

**Comment by afucher** [comment]
@karlhorky I'm using, and it works. You can check [here](https://github.com/afucher/afucher.github.io)

**Comment by karlhorky** [comment]
@afucher Your link is broken. I guess you mean https://github.com/afucher/afucher.github.io ?

Your code is using `gatsby-remark-external-links`:

https://github.com/afucher/afucher.github.io/blob/53ac0198fc5a164a69f569a9ce649934d8b5a9ab/gatsby-config.js#L42

My question is whether you have tried it with `remark-external-link`, which appears to be broken because Gatsby needs these wrapper packages :(

**Comment by afucher** [comment]
Yes @karlhorky , sry was on my phone, and the link was missing `https`.

I didn't try to make it works with `remark-external-link` because the docs (as you mentioned) said that I need the wrapper package.

**Comment by bathrobe** [comment]
Hi, I was wondering if anyone knew how to add `options` to `remark-external-links` when using `gatsby-plugin-mdx`. I am hoping to add the `content:` option in the original plugin (to add an "open in new window" icon at the end of each external link), but everywhere i look the remark plugin can only be successfully insert into `gatsby-plugin-mdx` with `require(remark-external-links)`, which says i can't pass an object.

**Comment by karlhorky** [comment]
Yeah, this should be documented better - I think this is kind of buried in the documentation somewhere in some Remark package.

The way that has worked for me is to use a 2-element array with the configuration in the second element (`gatsby-config.json`):

```js
module.exports = {
  plugins: [
    {
      resolve: `gatsby-plugin-mdx`,
      options: {
        remarkPlugins: [
          [require('remark-external-links'), {target: false}],
		],
      },
    },
  ],
}
```

**Comment by karlhorky** [comment]
Opened a PR to add this to the `gatsby-plugin-mdx` documentation: https://github.com/gatsbyjs/gatsby/pull/27301

**Comment by bathrobe** [comment]
@karlhorky thank you x100000. was REALLY scratching my head. the Hast dox are defo not quite beginner-friendly. For anyone else searching for new-window icon support, this was my solution! (i'm sure there's a cleaner way than having all this data in `gatsby-config` but i'm just happy it works)

```js
        remarkPlugins: [
          [
            require("remark-external-links"),
            {
              content: {
                type: "element",
                tagName: "svg",
                properties: {
                  xmlns: "http://www.w3.org/2000/svg",
                  width: "14",
                  height: "14",
                  viewBox: "0 0 24 24",
                  fill: "none",
                  stroke: "currentColor",
                  strokeWidth: "2",
                  strokeLinecap: "round",
                  strokeLinejoin: "round",
                  className: "feather feather-external-link",
                },
                children: [
                  {
                    type: "element",
                    tagName: "path",
                    properties: {
                      d:
                        "M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6",
                    },
                  },
                  {
                    type: "element",
                    tagName: "polyline",
                    properties: { points: "15 3 21 3 21 9" },
                  },
                  {
                    type: "element",
                    tagName: "line",
                    properties: {
                      x1: "10",
                      y1: "14",
                      x2: "21",
                      y2: "3",
                    },
                  },
                ],
              },
            },
          ],
        ],
```

**Comment by karlhorky** [comment]
One more trick, if you absolutely need a Gatsby remark plugin instead of a normal remark plugin, is to run the normal remark plugin through [`to-gatsby-remark-plugin`](https://www.npmjs.com/package/to-gatsby-remark-plugin) (really cool @kevin940726, thanks for this!!)

**Comment by wooorm** [comment]
That's cool @karlhorky, @kevin940726!
